keras_cv/models/efficientnet_v1.py

```diff
@@ -201,151 +201,252 @@ def correct_pad(inputs, kernel_size):
     )
 
 
-def EfficientNetBlock(
-    activation="swish",
-    drop_rate=0.0,
-    name="",
-    filters_in=32,
-    filters_out=16,
-    kernel_size=3,
-    strides=1,
-    expand_ratio=1,
-    se_ratio=0.0,
-    id_skip=True,
-):
+class EfficientNetBlock(layers.Layer):
```

Comment:
What's the reasoning behind making this a `Layer` instead of a function like `apply_efficientnet_block`, like we have for other models?

If it's practical to do so, we should keep this consistent with other models by using the `apply_xyz_block` function approach.

Reply:
The main purpose to use of `layer`  is to develop a architecture as a separate blocks as `EfficientNetV2` has been developed. 

In `EfficientNetV2`, the model is constructed as a different blocks connected together (`FusedMBConvBlock`, `MBConvBlock`). So I have thought `layer` implementation of `efficientnet_block` would make the model modular.  

Can you please have a suggestion on this ? If the implementations wants to be in functional type, I am ready to make changes.

Reply:
Gotcha -- thanks! 
I think for now let's make this a function, not a layer.

If we find that this is re-usable for other models, we can move it to `keras_cv/layers/` and use it from here

Reply:
Thank you for direction.

Successfully made the changes.